tests/generate_small_sample.py

```diff
@@ -0,0 +1,133 @@
+from test_main import TestFiles
+from pathlib import Path
+import hashlib
+
+from pysource_minimize import minimize
+import sys
+import subprocess as sp
+import textwrap
+import os
+import linecache
+from executing import Source
+from multiprocessing import Pool,get_context
+import tempfile
+import hashlib
+import time
+
+last_samples_dir = Path(__file__).parent / "last_samples"
+last_samples_dir.mkdir(exist_ok=True)
+samples_dir = Path(__file__).parent / "samples"
+
+
+# big_samples = (Path(__file__).parent / "samples").glob("*.py")
+# big_samples = list(Path.home().rglob("*.py"))
+small_samples = Path(__file__).parent / "small_samples"
+
+
+def big_samples():
+    yield from last_samples_dir.rglob("*.py")
+    yield from samples_dir.rglob("*.py")
+
+    hashes = set()
+    root = Path.home()
+    # root = Path(__file__).parent / "samples"
+    for p in root.rglob("*.py"):
+        try:
+            content = p.read_text()
+        except:
+            continue
+
+        if content.count("\n") > 100000:
+            continue
+
+        h = hashlib.sha256(content.encode()).hexdigest()
+        if h in hashes:
+            continue
+        hashes.add(h)
+        yield p
+
+
+def test_file(filename: Path):
+    sys.setrecursionlimit(3000)
+
+    if filename.read_text().count("\n") > 50000:
+        return True, filename
+
+    # clear caches
+    linecache.clearcache()
+    for cache_name in ("__source_cache_with_lines", "__executing_cache"):
```

Comment:
Why?

Reply:
i clear all caches because I experienced an increasing memory usage. Maybe not all this caches are a problem, but it was easier to clear them all. 
Coming back to your PC and figuring out that the test stopped, because python uses almost all your RAM was not very productive. 
I don't think that this cache issues are real problems because normal code will not analyse so many bytecodes.

Reply:
Makes sense, please add a comment to the code. I thought it was about the cache containing invalid data.